pkg/virt-launcher/virtwrap/live-migration-target.go

```diff
@@ -20,32 +20,107 @@
 package virtwrap
 
 import (
+	"encoding/xml"
 	"fmt"
 	"net"
 	"path/filepath"
 	"strconv"
+	"time"
 
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/log"
 
 	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
 	cmdv1 "kubevirt.io/kubevirt/pkg/handler-launcher-com/cmd/v1"
 	"kubevirt.io/kubevirt/pkg/hooks"
+	netvmispec "kubevirt.io/kubevirt/pkg/network/vmispec"
 	"kubevirt.io/kubevirt/pkg/util"
 	"kubevirt.io/kubevirt/pkg/util/net/ip"
 	migrationproxy "kubevirt.io/kubevirt/pkg/virt-handler/migration-proxy"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/converter"
 )
 
-func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance) error {
+func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance, options *cmdv1.VirtualMachineOptions) error {
+	interfaceMigrationOptions := options.InterfaceMigration
+	if err := l.reconnectGuestNics(vmi, interfaceMigrationOptions); err != nil {
```

Comment:
How about extracting the code that detects which ifaces should be "reconnected"
and then pass the relevant interfaces (if there are any) to `reconnectGuestNics`?

```
for ifaceName, migrationMehtod := range option.InterfaceMigration {
  if migraionMethod == "link-refresh" {
    if err := l.reconncetGuestInterface(vmi.metadata.name, ifaceName) {
        return err
    }
  }
}
```

Another option is to "transpose" `options.InterfaceMigration` so the migration-method is the key and the values are interfaces slice, it will make it easier to use.